databases/ClickHouse/Portfile

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github  1.0
+PortGroup           cmake   1.1
+
+github.setup        ClickHouse ClickHouse 22.4.2.1 v -stable
+
+description         ClickHouse® is a free analytics DBMS for big data
+
+long_description \
+    ClickHouse® is an open-source column-oriented database management \
+    system that allows generating analytical data reports in real-time.
+
+notes {
+    Install the ClickHouse-server port to run ClickHouse as a system service.
+}
+
+categories          databases
+maintainers         @slarew openmaintainer
+license             Apache-2
+platforms           darwin
+
+# clickhouse overrides the deployment target so match it
+macosx_deployment_target 10.15
+
+installs_libs       no
+
+# require vanilla clang
+compiler.whitelist  macports-clang-13 macports-clang-14
```

Comment:
Whitelisting compilers is seldom desirable. This list will become out of date over time. Instead, blacklist compilers that won't work so that MacPorts can pick another one from its list, which we will keep up to date. If Apple Clang won't work, `compiler.blacklist clang`, however also add a comment explaining why, since that's very unusual.